<sdf version="1.4">
  <world name="default">
    <scene>
      <ambient>0.8 0.8 0.8 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>true</shadows>
    </scene>

    <!-- A global light source -->
    <include>
      <uri>model://sun</uri>
      <pose> -2.0 2.0 2.0 0 0 0</pose>
    </include>



  <!-- Piso da sala - circuito -->
    <include>
      <uri>model://forca/pisosala</uri>
      <pose> 0 0 0 0 0 -1.57</pose>
    </include>

<!-- car -->
    <include>
      <uri>model://forca/objeto/car</uri>
      <pose> 3.17 -2.97 0.251305 0 0 0</pose>
    </include>

 <!-- horse -->
    <include>
      <uri>model://forca/objeto/horse</uri>
      <pose> -3.63 2.60 0.251305 0 0 0</pose>
    </include>
   

 <!-- cow -->
    <include>
      <uri>model://forca/objeto/cow</uri>
      <pose> 4.707 3.08 0.251305 0 0 0</pose>
    </include>

<!-- dog -->
    <include>
      <uri>model://forca/objeto/dog</uri>
      <pose> -5.176 -3.165 0.251305 0 0 -0.52</pose>
    </include>


<!-- caixa_01 -->
    <include>
      <uri>model://forca/caixas/caixa_1</uri>
      <pose> -6.05 4.86 0.251305 0 0 0</pose>
    </include>

<!-- caixa_02 -->
    <include>
      <uri>model://forca/caixas/caixa_2</uri>
      <pose> -5.51 4.86 0.251305 0 0 0</pose>
    </include>

<!-- caixa_03 -->
    <include>
      <uri>model://forca/caixas/caixa_3</uri>
      <pose> -4.95 4.85 0.251305 0 0 0</pose>
    </include>

<!-- caixa_04 -->
    <include>
      <uri>model://forca/caixas/caixa_4</uri>
      <pose> -6.13 3.17 0.251305 0 0 0</pose>
    </include>    

<!-- caixa_05 -->
    <include>
      <uri>model://forca/caixas/caixa_5</uri>
      <pose> -5.633 3.18 0.251305 0 0 0</pose>
    </include>

<!-- caixa_06 -->
    <include>
      <uri>model://forca/caixas/caixa_6</uri>
      <pose> -4.64 3.65 0.251305 0 0 0</pose>
    </include>

<!-- caixa_07 -->
    <include>
      <uri>model://forca/caixas/caixa_7</uri>
      <pose> -3.27 3.15 0.251305 0 0 0</pose>
    </include>

<!-- caixa_08 -->
    <include>
      <uri>model://forca/caixas/caixa_8</uri>
      <pose> -2.77 3.16 0.251305 0 0 0</pose>
    </include>

<!-- caixa_09 -->
    <include>
      <uri>model://forca/caixas/caixa_9</uri>
      <pose> -2.27 3.16 0.251305 0 0 0</pose>
    </include>

<!-- caixa_10 -->
    <include>
      <uri>model://forca/caixas/caixa_10</uri>
      <pose> -1.73 3.16 0.251305 0 0 0</pose>
    </include>

<!-- caixa_11 -->
    <include>
      <uri>model://forca/caixas/caixa_11</uri>
      <pose> -1.72 3.67 0.251305 0 0 0</pose>
    </include>

<!-- caixa_12 -->
    <include>
      <uri>model://forca/caixas/caixa_12</uri>
      <pose> -1.633 4.74 0.251305 0 0 0</pose>
    </include>

<!-- caixa_13 -->
    <include>
      <uri>model://forca/caixas/caixa_13</uri>
      <pose> -2.637 4.75 0.251305 0 0 0</pose>
    </include>

<!-- caixa_14 -->
    <include>
      <uri>model://forca/caixas/caixa_14</uri>
      <pose> -3.23 4.76 0.251305 0 0 0</pose>
    </include>

<!-- caixa_15 -->
    <include>
      <uri>model://forca/caixas/caixa_15</uri>
      <pose> 3.03 1.23 0.251305 0 0 0</pose>
    </include>

<!-- caixa_16 -->
    <include>
      <uri>model://forca/caixas/caixa_16</uri>
      <pose> 3.02 0.73 0.251305 0 0 0</pose>
    </include>

<!-- caixa_17 -->
    <include>
      <uri>model://forca/caixas/caixa_17</uri>
      <pose> 3.03 0.23 0.251305 0 0 0</pose>
    </include>

 <!-- caixa_18 -->
    <include>
      <uri>model://forca/caixas/caixa_18</uri>
      <pose> 3.04 -0.28 0.251305 0 0 0</pose>
    </include>   

<!-- caixa_19 -->
    <include>
      <uri>model://forca/caixas/caixa_19</uri>
      <pose> 3.04 -0.78 0.251305 0 0 0</pose>
    </include>


<!-- caixa_20 -->
    <include>
      <uri>model://forca/caixas/caixa_20</uri>
      <pose> 3.55 -0.77 0.251305 0 0 0</pose>
    </include>

<!-- coke -->
    <include>
      <uri>model:///forca/objeto/small_beer</uri>
      <pose> -1.74 4.23 0.251305 0 0 0</pose>
    </include> 


<!-- cilindro_01 -->
    <include>
      <uri>model://forca/objeto/cilindro_01</uri>
      <pose> 1.49 4.82 0.251305 0 0 0</pose>
    </include>

 <!--  creeper_50 -->
    <include>
      <uri>model://forca/creepers/creeper_50</uri>
      <pose> -4.79 2.256 0.1744 0 0 0.700</pose>
    </include>


 <!--  creeper_100 -->
    <include>
      <uri>model://forca/creepers/creeper_100</uri>
      <pose> -3.4 -0.32 0.1744 0 0 1.63</pose>
    </include>

 <!--  creeper_150 -->
    <include>
      <uri>model://forca/creepers/creeper_150</uri>
      <pose> -4.315 -3.09 0.1744 0 0 2.66</pose>
    </include>

 <!--  creeper_200 -->
    <include>
      <uri>model://forca/creepers/creeper_200</uri>
      <pose> 1.72 -0.063 0.1744 0 0 -1.64</pose>
    </include>





 <!-- Testing the automatic population of objects -->
    <population name="creeper_11_azul">
      <model name="creeper_11_azul">
        <include>
            <uri>model://forca/creepers/creeper_11_azul</uri>
        </include>
      </model>
      <pose> 1.71 -2.45 0 0 0 -1.17</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>

<!--  creeper_11_azul 
    <include>
      <uri>model://forca/creepers/creeper_11_azul</uri>
      <pose> 1.71 -2.45 0 0 0 -1.17</pose>
    </include> -->





<!--  creeper_11_laranja -->
    <population name="creeper_11_laranja">
      <model name="clone1">
        <include>
            <uri>model://forca/creepers/creeper_11_laranja</uri>
        </include>
      </model>
      <pose> 2.75 2.74 0 0 0 -2.72</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>

  <!--   <include>
      <uri>model://forca/creepers/creeper_11_laranja</uri>
      <pose> 4.96 -2.19 0 0 0 2.86</pose>
    </include> -->

 <!--  creeper_12_azul -->

     <population name="creeper_12_azul">
      <model name="creeper_12_azul">
        <include>
            <uri>model://forca/creepers/creeper_12_azul</uri>
        </include>
      </model>
      <pose> -1.26 -0.66 0 0 0 0</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>

   <!--   <include>
      <uri>model://forca/creepers/creeper_12_azul</uri>
      <pose> -1.26 -0.66 0 0 0 0</pose>
    </include> -->

<!--  creeper_13_verde -->

     <population name="creeper_13_verde">
       <model name="creeper_13_verde">
        <include>
            <uri>model://forca/creepers/creeper_13_verde</uri>
        </include>
      </model>
     <pose> 5.49 1.66 0 0 0 -1.57</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>

 <!--     <include>
      <uri>model://forca/creepers/creeper_13_verde</uri>
      <pose> 5.49 1.66 0 0 0 -1.57</pose>
    </include> -->

 <!-- creeper_21_verde -->

      <population name="creeper_21_verde">
      <model name="creeper_21_verde">
        <include>
            <uri>model://forca/creepers/creeper_21_verde</uri>
        </include>
      </model>
       <pose> -2.60 0.49 0 0 0 2.849</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>

  <!--    <include>
      <uri>model://forca/creepers/creeper_21_verde</uri>
       <pose> -2.60 0.49 0 0 0 2.849</pose>
    </include> -->

 <!-- creeper_21_laranja -->

      <population name="creeper_21_laranja">
      <model name="creeper_21_verde">
        <include>
            <uri>model://forca/creepers/creeper_21_laranja</uri>
        </include>
      </model>
      <pose> 3.29  2.82 0 0 0 -3.07</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>


<!--  creeper_21_laranja 
    <include>
      <uri>model://forca/creepers/creeper_21_laranja</uri>
      <pose> 3.29  2.82 0 0 0 -3.07</pose>
    </include> -->


<!--  creeper_22_azul -->


      <population name="creeper_22_azul">
  <model name="creeper_22_azul">
        <include>
            <uri>model://forca/creepers/creeper_22_azul</uri>
        </include>
      </model>
        <pose> -4.30 -1.94 s0 0 0 0.700</pose> 
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>


  <!--   <include>
      <uri>model://forca/creepers/creeper_22_azul</uri>
      <pose> -4.30 -1.94 s0 0 0 0.700</pose>    
      </include> -->
      
<-- creeper_23_verde -->

      <population name="creeper_23_verde">
      <model name="creeper_23_verde">
        <include>
            <uri>model://forca/creepers/creeper_23_verde</uri>
        </include>
      </model>
         <pose> -3.45 -2.88 0 0 0 1.04</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>


  <!--   <include>
      <uri>model://forca/creepers/creeper_23_verde</uri>
      <pose> -3.45 -2.88 0 0 0 1.04</pose>
    </include>  -->



 <!--  creeper_51_azul -->

      <population name="creeper_51_azul">
    <model name="creeper_51_azul">
        <include>
        <uri>model://forca/creepers/creeper_51_azul</uri>
        </include>
      </model>
      <pose> -5.80 3.69 0 0 0 0</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>


  <!--    <include>
      <uri>model://forca/creepers/creeper_51_azul</uri>
      <pose> -5.80 3.69 0 0 0 0</pose>
    </include> -->

 <!--  creeper_52_verde -->


      <population name="creeper_52_verde">
     <model name="creeper_52_verde">
        <include>
      <uri>model://forca/creepers/creeper_52_verde</uri>
        </include>
      </model>
      <pose> -2.30 3.69 0 0 0 0</pose>
      <box>
        <size>1 1 1</size>
      </box>
      <model_count>1</model_count>
      <distribution>
        <type>random</type>
      </distribution>
    </population>


   <!--   <include>
      <uri>model://forca/creepers/creeper_52_verde</uri>
      <pose> -2.30 3.69 0 0 0 0</pose>
    </include> -->

    <physics type="ode">
      <real_time_update_rate>1000.0</real_time_update_rate>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <ode>
        <solver>
          <type>quick</type>
          <iters>150</iters>
          <precon_iters>0</precon_iters>
          <sor>1.400000</sor>
          <use_dynamic_moi_rescaling>1</use_dynamic_moi_rescaling>
        </solver>
        <constraints>
          <cfm>0.00001</cfm>
          <erp>0.2</erp>
          <contact_max_correcting_vel>2000.000000</contact_max_correcting_vel>
          <contact_surface_layer>0.01000</contact_surface_layer>
        </constraints>
      </ode>
    </physics>
   
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose frame=''>0.238775 -6.8831 10.8485 -0 0.979642 1.5602</pose>
        <view_controller>orbit</view_controller>
        <projection_type>perspective</projection_type>
      </camera>
    </gui>

  </world>

</sdf>